### DISCUSSION TITLE
anyone use pycardano  in an android app?

### DISCUSSION THREAD

**Discussion by l-o-u-i-s-g** [Python-Cardano/pycardano#361] [Q&A]
I had this idea of making an airgap  cardano wallet  using pycardano  and the beeware/toga framework for building the app for Linux and android.  My preliminary tests work great on Linux  but fails on android.  The pycardano package gets install but the app fails to load.  It looks like pycardano  needs the files on disk at runtime?  I had a discussion with the developer of the framework , in short this is his response: 

`This might be due to how code is deployed on Android. In a desktop deployment, an installed package is a directory of files on disk; however, on Android, the code is contained in the APK file, which is effectively a zip. When running on Android, a custom loader is used to import code; this works fine for "standard" Python imports, but if cordarno is relying on assumptions about the availability of code on disk that aren't actually guaranteed by Python's import system, then it could be causing problems.`
full thread here: https://github.com/beeware/toga/discussions/2704

What is the docker dependency in pycardano for?   Does pycardano instantiate a docker image?  I could see problems here ...

**Comment by nielstron** [comment]
> What is the docker dependency in pycardano for? Does pycardano instantiate a docker image? I could see problems here ...

No PyCardano does not instantiate a docker image. This is only if the user wants to use an ogmios or cardano-cli instance that run in docker.
